File: http/src/main/java/webconnect/com/webconnect/observer/FailureLiveData.java
package webconnect.com.webconnect.observer;


import android.arch.lifecycle.MutableLiveData;

public class FailureLiveData extends MutableLiveData<String> {

    private static FailureLiveData sInstance;

    private FailureLiveData() {

    }

    public static FailureLiveData getInstance() {
        if (sInstance == null) {
            synchronized (FailureLiveData.class) {
                if (sInstance == null) {
                    sInstance = new FailureLiveData();
                }
            }
        }
        return sInstance;
    }
}

File: http/src/main/java/webconnect/com/webconnect/observer/FailureLiveData.kt
package webconnect.com.webconnect.observer;

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData


object FailureLiveData {
        internal val failure = MutableLiveData<String>()

        @JvmStatic
        fun getFailureLiveData(): LiveData<String> {
                return failure
        }
}

